### DISCUSSION TITLE
Help! Converting Pandas DataFrame to iterable Quotes

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#1165] [Help and support]
> [!TIP]
> For a working example using price quotes from a **pandas.DataFrame**, see our [_Replit_ of the Williams Fractal indicator](https://replit.com/@daveskender/Stock-Indicators-for-Python-Williams-Fractal).  If you're having general trouble getting setup, see our [QuickStart guide](https://github.com/DaveSkender/Stock.Indicators.Python.QuickStart#readme).
>
> The most common problem users have, when dealing with DataFrame and quote conversions, is with compatibility between their own environment [Python locale settings](https://docs.python.org/3/library/locale.html) and source quote date/time and numeric data formats.

Also, checkout these issues from others, and how they've resolved the issue:

- https://github.com/DaveSkender/Stock.Indicators/discussions/1164
- https://github.com/DaveSkender/Stock.Indicators/discussions/1148
- https://github.com/DaveSkender/Stock.Indicators.Python/issues/346
- https://github.com/DaveSkender/Stock.Indicators.Python/issues/348
- https://github.com/DaveSkender/Stock.Indicators/discussions/1016
- https://github.com/DaveSkender/Stock.Indicators/discussions/1020
- https://github.com/DaveSkender/Stock.Indicators/discussions/781
- https://github.com/DaveSkender/Stock.Indicators.Python/issues/297
- https://github.com/DaveSkender/Stock.Indicators.Python/issues/295
- https://github.com/DaveSkender/Stock.Indicators.Python/issues/228

If these materials still don't help you resolve your problem, please export your DataFrame data using [.to_csv](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_csv.html) and then upload it here or [in a new Issue](https://github.com/DaveSkender/Stock.Indicators.Python/issues/new?labels=user%20support), without altering it, so we can inspect it for data compatibility issues.

```python
df.to_csv('quotes.csv',index=False)
```

**Comment by kathiebrady** [comment]
> Tip
> 
> For a working example using price quotes from a **pandas.DataFrame**, see our [_Replit_ of the Williams Fractal indicator](https://replit.com/@daveskender/Stock-Indicators-for-Python-Williams-Fractal). If you're having general trouble getting setup, see our [QuickStart guide](https://github.com/DaveSkender/Stock.Indicators.Python.QuickStart#readme).
> 
> The most common problem users have, when dealing with DataFrame and quote conversions, is with compatibility between their own environment [Python locale settings](https://docs.python.org/3/library/locale.html) and source quote date/time and numeric data formats.
> 
> Also, checkout these issues from others, and how they've resolved the issue:
> 
> * [`FormatException` when converting pandas.DataFrame to iterable quotes #1164](https://github.com/DaveSkender/Stock.Indicators/discussions/1164)
> * [Volatility stop problem #1148](https://github.com/DaveSkender/Stock.Indicators/discussions/1148)
> * [help: getting empty values with Williams Fractal facioquo/stock-indicators-python#346](https://github.com/facioquo/stock-indicators-python/issues/346)
> * [Format of quotes problem facioquo/stock-indicators-python#348](https://github.com/facioquo/stock-indicators-python/issues/348)
> * [Need help to use indicators, how to obtain or convert quotes #1016](https://github.com/DaveSkender/Stock.Indicators/discussions/1016)
> * [PYTHON - df conversion with quote #1020](https://github.com/DaveSkender/Stock.Indicators/discussions/1020)
> * [How do I get Binance data into proper Quote format (in Python)? #781](https://github.com/DaveSkender/Stock.Indicators/discussions/781)
> * [Does the class Quote only accept Daily-Resolution data? facioquo/stock-indicators-python#297](https://github.com/facioquo/stock-indicators-python/issues/297)
> * [ZigZag Fatal error. System.AccessViolationException: Attempted to read or write protected memory. facioquo/stock-indicators-python#295](https://github.com/facioquo/stock-indicators-python/issues/295)
> * [What´s the right way to create the iterable "quotes" list from a dataframe? facioquo/stock-indicators-python#228](https://github.com/facioquo/stock-indicators-python/issues/228)
> 
> If these materials still don't help you resolve your problem, please export your DataFrame data using [.to_csv](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_csv.html) and then upload it here or [in a new Issue](https://github.com/DaveSkender/Stock.Indicators.Python/issues/new?labels=user%20support), without altering it, so we can inspect it for data compatibility issues.
> 
> ```python
> df.to_csv('quotes.csv',index=False)
> ```
